src/rhash/hash.c

```diff
@@ -1250,7 +1295,16 @@ static int rc_hash_nintendo_disc(char hash[33], const rc_hash_iterator_t* iterat
     {
       success = rc_hash_wii(&md5, iterator, file_handle);
     } else {
-      success = rc_hash_iterator_error(iterator, "Not a supported Nintendo disc");
+      /* Check WiiWare */
+      rc_file_seek(iterator, file_handle, 0x04, SEEK_SET);
+      rc_file_read(iterator, file_handle, quad_buffer, 2);
+      if (quad_buffer[0] == 'I' && quad_buffer[1] == 's')
```

Comment:
Consider also validating buffer[2] and [3], which I believe both should be 0x00 per the linked document. This will reduce the likelihood of false positives.